src/arrayMethodSort.js

```diff
@@ -1,11 +1,30 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compare =
+      compareFunction ||
+      function (x, y) {
+        if (String(x) > String(y)) {
+          return 1;
+        }
+
+        if (String(x) < String(y)) {
+          return -1;
+        }
+
+        return 0;
```

Comment:
The default compare function works well for string comparison. However, for numeric sorts, it might not sort numbers as expected since it converts numbers to strings before comparison. If you're planning to use this function for sorting numbers, consider implementing a numeric comparison as well.